python/vineyard/data/dataframe.py

```diff
@@ -51,7 +52,12 @@ def dataframe_resolver(obj, resolver):
     index_size = 0
     for idx, name in enumerate(columns):
         np_value = resolver.run(obj.member('__values_-value-%d' % idx))
-        blocks.append(Block(np.expand_dims(np_value, 0), slice(idx, idx + 1, 1)))
+        if LooseVersion(pd.__version__) < LooseVersion('1.2.0'):
+            kw = dict()
+        else:
+            # ndim: 1 for SingleBlockManager/Series, 2 for BlockManager/DataFrame
+            kw = {'ndim': 2}
+        blocks.append(Block(np.expand_dims(np_value, 0), slice(idx, idx + 1, 1), **kw))
```

Comment:
Why don't always pass `ndim=2`? It should work fine for both versions and don't need additional version check. When pandas < 1.2.0 the signature is `ndim=None`, and after 1.2.0 they just drop the default None.

Reply:
or more intelligently, use `value = np.expand_dims(np_value, 0); ndim=value.ndim`

Reply:
Fixed. Thanks!